homeassistant/components/econet/config_flow.py

```diff
@@ -0,0 +1,62 @@
+"""Config flow to configure the EcoNet component."""
+from pyeconet import EcoNetApiInterface
+from pyeconet.errors import InvalidCredentialsError, PyeconetError
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.const import CONF_EMAIL, CONF_PASSWORD
+
+from .const import DOMAIN  # pylint: disable=unused-import
+
+
+class EcoNetFlowHandler(config_entries.ConfigFlow, domain=DOMAIN):
+    """Handle an EcoNet config flow."""
+
+    VERSION = 1
+    CONNECTION_CLASS = config_entries.CONN_CLASS_CLOUD_POLL
+
+    def __init__(self):
+        """Initialize the config flow."""
+        self.data_schema = vol.Schema(
+            {
+                vol.Required(CONF_EMAIL): str,
+                vol.Required(CONF_PASSWORD): str,
+            }
+        )
+
+    async def _show_form(self, error=None):
+        """Show the form to the user."""
+        return self.async_show_form(
+            step_id="user",
+            data_schema=self.data_schema,
+            errors={"base": error} if error else {},
+        )
+
+    async def async_step_import(self, import_config):
+        """Import a config entry from configuration.yaml."""
+        return await self.async_step_user(import_config)
+
+    async def async_step_user(self, user_input=None):
+        """Handle the start of the config flow."""
+        if not user_input:
+            return await self._show_form()
+
+        await self.async_set_unique_id(user_input[CONF_EMAIL])
+        self._abort_if_unique_id_configured()
+
+        try:
+            await EcoNetApiInterface.login(
+                user_input[CONF_EMAIL], user_input[CONF_PASSWORD]
+            )
+        except InvalidCredentialsError:
+            return await self._show_form(error="invalid_auth")
```

Comment:
Just store the errors in a variable and after the except blocks check the errors and if it's truthy show the form.

Reply:
done